test/gtest/uct/uct_test.cc

```diff
@@ -1250,17 +1250,20 @@ uct_test::entity::connect_to_sockaddr(unsigned index,
                         UCT_EP_PARAM_FIELD_SOCKADDR_CONNECT_CB_CLIENT |
                         UCT_EP_PARAM_FIELD_SOCKADDR_DISCONNECT_CB;
 
-    params.user_data          = new sockaddr_user_data(test, this, index);
-    params.cm                 = m_cm;
-    params.sockaddr           = &ucs_remote_addr;
-    params.sockaddr_cb_flags  = UCT_CB_FLAG_ASYNC;
-    params.cm_resolve_cb      = resolve_cb;
-    params.sockaddr_cb_client = connect_cb;
-    params.disconnect_cb      = disconnect_cb;
-    status                    = uct_ep_create(&params, &ep);
+    sockaddr_user_data *sa_data = new sockaddr_user_data(test, this, index);
```

Comment:
maybe pass user_data as argument to connect_to_sockaddr() [instead of passing this]? 
so the test itself would be the owner of user_data, and not need for add_user_data to be virtual function